src/client/nuclearnet/web_socket_proxy_nuclearnet_client.ts

```diff
@@ -31,11 +32,11 @@ export class WebSocketProxyNUClearNetClient implements NUClearNetClient {
   }
 
   static of() {
-    const uri = `${document.location.origin}/nuclearnet`
-    return new WebSocketProxyNUClearNetClient(WebSocketClient.of(uri, {
+    return new WebSocketProxyNUClearNetClient(WebSocketClient.of(document.location.origin, {
       upgrade: false,
       transports: ['websocket'],
-    }))
+      parser: NUClearProxyParser,
+    } as any))
```

Comment:
Is this because have old types?

Reply:
Yes, one day when the socket.io types get updated to 2.0 we might not need this.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23776 is the closest but got closed for some reason.